apps/projects/app/components/Panel/ReviewApplication/ReviewApplication.js

```diff
@@ -34,12 +36,32 @@ const ReviewApplication = ({ issue, requestIndex, readOnly }) => {
   } = useAragonApi()
   const { closePanel } = usePanelManagement()
   const theme = useTheme()
-
   const [ feedback, setFeedback ] = useState('')
-  const [ index, setIndex ] = useState(requestIndex)
 
-  const updateFeedback = e => setFeedback(e.target.value)
+  const canReview = issue.workStatus !== 'fulfilled' && !readOnly
+
+  const typesToShow = []
+  if (issue.requestsData.findIndex(request => !('review' in request)) !== -1)
+    typesToShow.push(canReview ? 'Available for review' : 'Unreviewed')
+  if (issue.requestsData.findIndex(request => 'review' in request) !== -1)
+    typesToShow.push('Reviewed')
+
+  let indexTypeSelected = 0
+  if ('review' in issue.requestsData[requestIndex])
+    indexTypeSelected = (typesToShow[0] === 'Reviewed') ? 0 : 1
+
+  const [ indexType, setIndexType ] = useState(indexTypeSelected)
```

Comment:
I find it strange that it's possible that `indexTypeSelected` will change from one render to the next, but the state set up here will not change with it! As in, if on first render `indexTypeSelected` is `1`, then `indexType` will be initialized to `1`. But if on second render `indexTypeSelected` is `0`, `indexType` will still be `1`, because it was already initialized and we're not calling `setState`.

I'm wondering if the calculation of `indexTypeSelected` and `typesToShow` could be moved to an external function or maybe a hook, to be used like:

```js
const [ filters, selectedFilter ] = useReviewFilters(issue.requestsData)
```

Then they could be memoized and only change if `issue.requestsData` changes.

Reply:
Hook created and reused in both panels.